plutus-core/plutus-ir/src/PlutusIR/Purity.hs

```diff
@@ -75,41 +76,47 @@ instance Pretty WorkFreedom where
   pretty MaybeWork = "maybe work?"
   pretty WorkFree  = "work-free"
 
--- | Either the "next" term to be evaluated, along with its 'Purity' and 'WorkFreedom',
--- or we don't know what comes next.
-data EvalTerm tyname name uni fun a =
-  Unknown
+{- | Either the "next" term to be evaluated, along with its 'Purity' and 'WorkFreedom',
```

Comment:
The more this happens the more I dislike our current approach of "people can just reformat things to their personal preference". It's terrible. _Please_ can we decide to either use `fourmolu` consistently or just not reformat things.

Reply:
At the moment nothing stops someone just reformatting this back a different way with "Chore: reformatting" :(

Reply:
> stylish-haskell and it seems to format new code in the same way the existing code in the module is formatted

Two reasons:
* we already enforce stylish-haskell (git hooks) .
* stylish-haskell is very liberal: it does change the imports and aligns vertical arrows, equals, etc. It won't change indentation.

By the way, I have configured local formatting such that  `fourmolu` -produced output is piped to the `stylish-haskell`: their combination leads to a least change.

Reply:
> A thing that is decided by a fourmolu

Nothing in our style guide or contributing guide even says that we use fourmolu at all.

> By the way, I have configured local formatting such that fourmolu -produced output is piped to the stylish-haskell: their combination leads to a least change.

:scream: 

> did the team try to use fourmolu at some point? What didn't they like about it?

@effectfully is the main opponent, you can search in slack for some discussion.

Reply:
@ana-pantilie see [this](https://github.com/IntersectMBO/plutus/pull/5761#pullrequestreview-1863517330) for example.

Reply:
> see https://github.com/IntersectMBO/plutus/pull/5761#pullrequestreview-1863517330 for example.

I initially read the first diff as fourmolu removing the `INLINEABLE` pragmas which got me very scared, but I see it's only adding a newline between the pragma and the type declaration, right? I looked up the docs and `INLINEABLE` or `INLINE` can be added anywhere in the module I believe, https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/pragmas.html#inlinable-pragma.
The other changes you pointed out aren't really dangerous, are they? If they're not, although I agree that they're ugly, I think they're not _that bad_ once you consider the bigger picture. Don't hate me, but I think that as long as these "formatting bugs" are cosmetic and relatively infrequent it's a small price to pay for having consistent automated formatting.